doc/ref/plot_options/data.ipynb

```diff
@@ -19,7 +19,7 @@
    "source": [
     "## `by`\n",
     "\n",
-    "Text TBD."
+    "The `by` option allows you to group your data based on one or more categorical variables. By specifying a column name (or a list of column names) with `by`, the plot automatically separates the data into groups. This makes it easier to compare different subsets of your data in a single visualization. For instance, in the penguin dataset, grouping by 'species' column creates separate overlays (or subplots when using `subplots=True`) for each species."
```

Comment:
hvPlot users are often confused with `by`, `groupby`, and `color`. Let's keep in mind we should make it clear in the reference how they differ (I also think we will have specific how-tos for that).